.github/workflows/zizmor.yml

```diff
@@ -0,0 +1,24 @@
+name: zizmor
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["**"]
+
+permissions: {}
+
+jobs:
+  zizmor:
+    name: Run zizmor
+    runs-on: ubuntu-latest
+    permissions:
+      security-events: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          persist-credentials: false
+
+      - name: Run zizmor
```

Comment:
I try to keep the amount of workflows small. As this is a check that should run on every PR/merge please add it to the `lint` job in the `test.yml` workflow: https://github.com/astral-sh/setup-uv/blob/26cf676705ba2848bf75ebd447bb297a435baee3/.github/workflows/test.yml#L19-L39